src/external/spglib.jl

```diff
@@ -162,3 +162,53 @@ function spglib_get_stabilized_reciprocal_mesh(kgrid_size, rotations::Vector;
 
     return n_kpts, Int.(mapping), [Vec3{Int}(grid_address[:, i]) for i in 1:nkpt]
 end
+
+struct SpglibDataset
+    spacegroup_number::Cint
+    hall_number::Cint
+    international_symbol::NTuple{11, Cchar}
+    hall_symbol::NTuple{17, Cchar}
+    choice::NTuple{6, Cchar}
+    transformation_matrix::NTuple{9, Cdouble}
+    origin_shift::NTuple{3, Cdouble}
+    n_operations::Cint
+    rotations::Ptr{NTuple{9, Cint}}
+    translations::Ptr{NTuple{3, Cdouble}}	
+    n_atoms::Cint
+    wyckoffs::Ptr{Cint}
+    site_symmetry_symbols::Ptr{NTuple{7, Cchar}}
+    equivalent_atoms::Ptr{Cint}
+    crystallographic_orbits::Ptr{Cint}
+    primitive_lattice::NTuple{9, Cdouble}
+    mapping_to_primitive::Ptr{Cint}
+    n_std_atoms::Cint
+    std_lattice::NTuple{9, Cdouble}
+    std_types::Ptr{Cint}
+    std_positions::Ptr{NTuple{3, Cdouble}}
+    std_rotation_matrix::NTuple{9, Cdouble}
+    std_mapping_to_primitive::Ptr{Cint}
+    pointgroup_symbol::NTuple{6, Cchar}
+end
+
+
+function spglib_get_dataset(lattice, atoms; tol_symmetry=1e-5)
+    # Convert lattice and atoms to spglib
+    spg_lattice = copy(Matrix{Float64}(lattice)')
+    spg_positions, spg_numbers, _ = spglib_atoms(atoms)
+    # Get the spglib dataset 
+    spglib_dataset  = ccall((:spg_get_dataset, SPGLIB), Ptr{SpglibDataset}, 
+	(Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cint}, Cint, Cdouble),
+	spg_lattice, spg_positions, spg_numbers, 
+	length(spg_numbers), tol_symmetry)
+    return unsafe_load(spglib_dataset)
+end
+
+function spglib_get_std_lattice(lattice, atoms)
+    spglib_dataset = spglib_get_dataset(lattice, atoms)
+    return transpose(reshape(collect(spglib_dataset.std_lattice), 3, 3))
+end
+
+function spglib_get_spacegroup(lattice, atoms)
+    spglib_dataset = spglib_get_dataset(lattice, atoms)
+    return spglib_dataset.spacegroup_number
```

Comment:
No return needed. The return of the last statement is automatic in julia.

Also: You should convert from `Cint` to `Int` to avoid bleeding out c-types beyond `spglib.jl`.